test/application/use.js

```diff
@@ -70,60 +65,50 @@ describe('app.use(fn)', () => {
 
     const server = app.listen();
 
-    request(server)
-      .get('/')
-      .expect(404)
-      .end(err => {
-        if (err) return done(err);
-        calls.should.eql([1, 2, 3, 4, 5, 6]);
-        done();
-      });
+    const res = await request(server, '/');
+    expect(res.status).toBe(404);
+    expect(calls).toEqual([1, 2, 3, 4, 5, 6]);
   });
 
   // https://github.com/koajs/koa/pull/530#issuecomment-148138051
-  it('should catch thrown errors in non-async functions', done => {
+  it('should catch thrown errors in non-async functions', async () => {
     const app = new Koa();
 
     app.use(ctx => ctx.throw('Not Found', 404));
 
-    request(app.listen())
-      .get('/')
-      .expect(404)
-      .end(done);
+    const res = await request(app, '/');
+    expect(res.status).toBe(404);
   });
 
-  it('should accept both generator and function middleware', done => {
+  it('should accept both generator and function middleware', async () => {
     process.once('deprecation', () => {}); // silence deprecation message
     const app = new Koa();
 
     app.use((ctx, next) => { return next(); });
     app.use(function * (next){ this.body = 'generator'; });
 
-    request(app.listen())
-      .get('/')
-      .expect(200)
-      .expect('generator', done);
+    const res = await request(app, '/');
+    expect(res.status).toBe(200);
+    expect(await res.text()).toBe('generator');
   });
 
   it('should throw error for non function', () => {
     const app = new Koa();
 
-    [null, undefined, 0, false, 'not a function'].forEach(v => (() => app.use(v)).should.throw('middleware must be a function!'));
+    [null, undefined, 0, false, 'not a function'].forEach(v => expect(() => app.use(v)).toThrow('middleware must be a function!'));
   });
 
-  it('should output deprecation message for generator functions', done => {
-    process.once('deprecation', message => {
-      assert(/Support for generators will be removed/.test(message));
-      done();
-    });
-
+  it('should output deprecation message for generator functions', () => {
+    const warningPromise = eventToPromise(process, 'deprecation')
```

Comment:
With the original syntax, the asserts for the result of the request were run after the test has finished. I don't know if they would be ignored or result in a strange error, but I do know that it wouldn't work properly. I can return it to its original syntax and remove the request asserts though.